cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -233,6 +235,34 @@ def _set_default_instance_role(self):
             ],
         }
 
+        if build.components:
+            for custom_component in build.components:
+                # Check custom component is script, and the url is S3 url
+                if custom_component.type == "script" and (
+                    custom_component.value.startswith("s3") or check_https_url_type(custom_component.value)
+                ):
+                    bucket_info = get_bucket_object_name(custom_component.value)
```

Comment:
what if the script is specified through a https url which is not hosted in S3? e.g. a script could be published through GitHub. We should pass the https url as it is, without doing any other background action

Reply:
done